src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,35 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo;
+  closeModal: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, closeModal }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const {
+    id, title, userId, completed,
+  } = selectedTodo;
+
+  useEffect(() => {
+    getUser(userId)
+      .then(selectedUser => {
+        setUser(selectedUser);
+      })
+      .catch(() => {
+        throw new Error('User not found');
+      });
+  }, []);
```

Comment:
Make correct dependencies 
```suggestion
  }, [userId]);
```